### PR TITLE
add governed owner

### PR DESCRIPTION
**Pull request by xiaozhou**
governed-owner to do infrequent sgn/cbridge/message/etc owner operations, relatively prefer easy-to-use over gas-efficiency.

Basic operations: 
- Customized `createXxxProposal` or generic `createProposal`
- `voteProposal`
- `executeProposal`

Will add more tests before merging.

### CODE REVIEW

File: contracts/governed-owner/SimpleGovernance.sol [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    constructor() {
+        deployer = msg.sender;
+    }
+
+    /*********************************
+     * External and Public Functions *
+     *********************************/
+
+    // initialize configs
+    function init(
```
Comment (Dominator008) [inline contracts/governed-owner/SimpleGovernance.sol:89]: Is this meant to be upgradable?

File: contracts/governed-owner/SimpleGovernance.sol [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+        address[] memory _voters,
+        uint256[] memory _powers,
+        address[] memory _proxies,
+        uint256 _activePeriod,
+        uint256 _quorumThreshold,
+        uint256 _fastPassThreshold
+    ) external {
+        require(msg.sender == deployer, "only deployer can init settings");
+        require(voters.length == 0, "alreayd initiated");
```
Comment (Dominator008) [inline contracts/governed-owner/SimpleGovernance.sol:98]: Typo

File: contracts/governed-owner/proxies/OwnerProxyBase.sol [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity >=0.8.0;
+
+import "../SimpleGovernance.sol";
+
+abstract contract OwnerProxyBase {
+    SimpleGovernance public gov;
+
+    function initGov(SimpleGovernance _gov) public {
```
Comment (Dominator008) [inline contracts/governed-owner/proxies/OwnerProxyBase.sol:10]: Anyone can init?

File: contracts/governed-owner/SimpleGovernance.sol (hunk 1 above)
Comment (xiaozhou) [reply contracts/governed-owner/SimpleGovernance.sol:89]: intend was to make inheritance simpler when the design pattern was different. Will move this to constructor

File: contracts/governed-owner/proxies/OwnerProxyBase.sol (hunk 3 above)
Comment (xiaozhou) [reply contracts/governed-owner/proxies/OwnerProxyBase.sol:10]: no security risk, but will add restriction to deployer only

**Pull request merged**